src/server/pps/server/api_server.go

```diff
@@ -2272,32 +2351,7 @@ func (a *apiServer) deletePipeline(ctx context.Context, pipeline *ppsclient.Pipe
 	var eg errgroup.Group
 	for _, jobInfo := range jobInfos.JobInfo {
 		jobInfo := jobInfo
-		eg.Go(func() error {
-			if err = a.kubeClient.Extensions().Jobs(a.namespace).Delete(jobInfo.JobID, nil); err != nil {
-				// we don't return on failure here because jobs may get deleted
-				// through other means and we don't want that to prevent users from
-				// deleting pipelines.
-				protolion.Errorf("error deleting job %s: %s", jobInfo.JobID, err.Error())
-			}
-			pods, jobPodsErr := a.jobPods(client.NewJob(jobInfo.JobID))
-			for _, pod := range pods {
-				if err = a.kubeClient.Pods(a.namespace).Delete(pod.Name, nil); err != nil {
-					// we don't return on failure here because pods may get deleted
-					// through other means and we don't want that to prevent users from
-					// deleting pipelines.
-					protolion.Errorf("error deleting pod %s: %s", pod.Name, err.Error())
-				}
-			}
-			if jobPodsErr != nil {
-				return jobPodsErr
-			}
-
-			// Remove the chunks for this job
-			_, err := persistClient.DeleteChunksForJob(ctx, &ppsclient.Job{
-				ID: jobInfo.JobID,
-			})
-			return err
-		})
+		eg.Go(func() error { return a.deleteJob(ctx, jobInfo) })
```

Comment:
Yea! 

I think there's more we can get rid of. It seems like k8s has an API to delete dependent pods when you're deleting a higher level object. For example here look at how I delete the `ReplicationController` in `deleteJob()` ... I think we can use the same thing for k8s 'Job' deletion instead of the pod deletion code below that. But I'll play w that in a separate PR